test/parallel/market.sellAsset.test.ts

```diff
@@ -0,0 +1,181 @@
+    sudo,
+  );
+});
+
+test("Sell asset - Only sold asset ( USDC ) in the wallet (Stable pool, amount> threshold)", async () => {
+  const poolEvent = await Sudo.batchAsSudoFinalized(
+    Market.createPool(
+      firstCurrency,
+      threshold.muln(5),
+      secondCurrency,
+      threshold.muln(5),
+      "StableSwap",
+    ),
+    Market.createPool(
+      GASP_ASSET_ID,
+      threshold.muln(5),
+      firstCurrency,
+      threshold.muln(5),
+      "StableSwap",
+    ),
+    Market.createPool(
+      GASP_ASSET_ID,
+      threshold.muln(5),
+      secondCurrency,
+      threshold.muln(5),
+      "StableSwap",
+    ),
+  );
+
+  await updateFeeLockMetadata(sudo, null, null, null, [[firstCurrency, true]]);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(firstCurrency, testUser, threshold.muln(5)),
+  );
+
+  const liqId = await getPoolIdFromEvent(poolEvent);
+
+  const poolBalance = await getBalanceOfPool(firstCurrency, secondCurrency);
+
+  const sellPrice = calculate_sell_price_local(
+    poolBalance[0],
+    poolBalance[1],
+    threshold.add(threshold.divn(2)),
+  );
+
+  const events = await signTx(
+    api,
+    Market.sellAsset(
+      liqId,
+      firstCurrency,
+      secondCurrency,
+      threshold.add(threshold.divn(2)),
+    ),
+    testUser.keyRingPair,
+  );
+  const res = getEventResultFromMangataTx(events);
+  expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  const filterEvents = events.filter((x) => x.method === "Endowed");
+  expect(filterEvents[2].eventData[2].data).bnEqual(sellPrice);
+});
+
+test("Sell asset - Only sold asset ( USDC ) in the wallet (Xyk pool, amount> threshold)", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Market.createPool(
+      firstCurrency,
+      threshold.muln(2),
+      secondCurrency,
+      threshold.muln(2),
+    ),
+    Market.createPool(
+      GASP_ASSET_ID,
+      threshold.muln(2),
+      firstCurrency,
+      threshold.muln(2),
+    ),
+    Market.createPool(
+      GASP_ASSET_ID,
+      threshold.muln(2),
+      secondCurrency,
+      threshold.muln(2),
+    ),
+  );
+
+  await updateFeeLockMetadata(sudo, null, null, null, [[firstCurrency, true]]);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(firstCurrency, testUser, threshold.muln(2)),
+  );
+
+  const liqId = await getLiquidityAssetId(firstCurrency, secondCurrency);
+
+  const poolBalance = await getBalanceOfPool(firstCurrency, secondCurrency);
+
+  const sellPrice = calculate_sell_price_local(
+    poolBalance[0],
+    poolBalance[1],
+    threshold.add(threshold.divn(2)),
+  );
+
+  const events = await signTx(
+    api,
+    Market.sellAsset(
+      liqId,
+      firstCurrency,
+      secondCurrency,
+      threshold.add(threshold.divn(2)),
+    ),
+    testUser.keyRingPair,
+  );
+  const res = getEventResultFromMangataTx(events);
+  expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  const filterEvents = events.filter((x) => x.method === "Endowed");
```

Comment:
why are you checking this endowed? 
why not check the user's wallet?

Reply:
Yeah, you're right. Changed to checking wallet